### DISCUSSION TITLE
Why use signed 64-bit for timestamp

### DISCUSSION THREAD

**Discussion by BratSinot** [time-rs/time#464] [Question]
Greetings!

I don't get it, why `i64` for Unix timestamp? Unix timestamp can't go before 1970 by definition. What the point?

**Comment by jhpratt** [comment]
Negative timestamps are for times before 1970. This is common practice.

**Reply by BratSinot** [reply]
> Negative timestamps are for times before 1970. This is common practice.

> Unix timestamp can't go before 1970 by definition.

Function name is `unix_timestamp`, so question still here )

**Reply by jhpratt** [reply]
To put it simply, your assertion is not the case.
